TestMissedRevenueHeatMapGenerator.py

```diff
@@ -0,0 +1,187 @@
+        hours_open = 8
+        journal = [{'Branch':self.branch, 'Date':self.sale_date_1, 'Increment':-6, 'Sku':self.sku, 'Tag':'sale'},
+                   {'Branch':self.branch, 'Date':self.purchase_date_1, 'Increment':2, 'Sku':self.sku, 'Tag':'purchase'},
+                   {'Branch':self.branch, 'Date':self.sale_date_2, 'Increment':-7, 'Sku':self.sku, 'Tag':'sale'}]
+        
+        #increment of sales is negative
+        #sales = 6 + 7
+        sales = 13
+        expected_average_rate_of_sale_per_hour = sales / hours_open
+        calculated_average_rate_of_sale_per_hour = MR.calculate_average_rate_of_sale_per_hour(self.sku, self.branch, journal, hours_open)
+        self.assertEqual("calculate_average_rate_of_sale_per_hour() returned incorrect value", calculated_average_rate_of_sale_per_hour, expected_average_rate_of_sale_per_hour)
+
+    def test_calculate_lost_qty_sales_for_a_period(self):
+        '''Test the calculated quantity of lost sales for a period.'''
+        
+        #TODO: reuse to calculate monthly/quaterly/yearly run rate
+        journal = [{'Branch':self.branch, 'Date':self.time_branch_opens, 'Increment':10, 'Sku':self.sku, 'Tag':'purchase'},
+                   {'Branch':self.branch, 'Date':self.sale_date_1, 'Increment':-10, 'Sku':self.sku, 'Tag':'sale'},
+                   {'Branch':self.branch, 'Date':self.purchase_date_1, 'Increment':1, 'Sku':self.sku, 'Tag':'purchase'}]
+
+        #TODO: Create a function to calculate amount of hours in a period
+        time_period = (self.time_branch_opens, self.time_branch_closes)
+        amount_of_hours = 8
+        #stock level fell below/equal danger level for 1 hour during this time period
+        danger_period_within_time_period = 1
+        amount_of_sales = 10
+        # average_rate_of_sale_per_hour = 10 sales / 8 hours
+        average_rate_of_sale_per_hour = amount_of_sales / amount_of_hours
+        qty_of_lost_sales = danger_period_within_time_period * average_rate_of_sale_per_hour
+    
+        calculated_qty_of_lost_sales = MR.test_calculate_lost_qty_sales_for_a_period(self.sku, self.branch, journal, time_period, average_rate_of_sale_per_hour)
+        self.assertEqual("calculated_qty_of_lost_sales() returned incorrect value", calculated_qty_of_lost_sales, qty_of_lost_sales)
+
+    def test_calculate_run_rate_for_a_week(self):
+        '''Test the calculated run rate of lost sales for a period.
+           Run rate is used to extrapolate data, in this case amount of sales, to predict quantity of future sales'''
+        #define one week period
+        period = (self.time_branch_opens, self.one_week_later)
+        
+        journal = [{'Branch':self.branch, 'Date':self.sale_date_1, 'Increment':-6, 'Sku':self.sku, 'Tag':'sale'},
+                   {'Branch':self.branch, 'Date':self.sale_date_2, 'Increment':-10, 'Sku':self.sku, 'Tag':'sale'}]
+
+        #calculate average_sales_per_hour for one day: 16 sales / 8 hours
+        average_sales_per_hour = 2
```

Comment:
this is a function of time and interval. The spec specifically asks to cater for different time intervals. Have you considered this? How would you make it work for hour/day/week/month/quarter/year?